editor/src/messages/tool/tool_messages/select_tool.rs

```diff
@@ -610,30 +860,35 @@ impl Fsm for SelectToolFsmState {
 					if !tool_data.is_dragging && input.keyboard.get(remove_from_selection as usize) && tool_data.layer_selected_on_start.is_none() {
 						let quad = tool_data.selection_quad();
 						let intersection = document.document_legacy.intersects_quad_root(quad, render_data);
-						if let Some(intersect_layer_path) = intersection.last() {
-							let replacement_selected_layers = document.selected_layers().filter(|&layer| layer != intersect_layer_path).map(|path| path.to_vec()).collect();
-							responses.push_back(DocumentMessage::SetSelectedLayers { replacement_selected_layers }.into());
-						}
+						let temp_intersection = intersection.clone();
```

Comment:
Is a new variable and clone necessary here?

Reply:
Without the temp variable I got an error shown in the picture
<img width="764" alt="Screen Shot 2023-03-10 at 1 11 42 PM" src="https://user-images.githubusercontent.com/57922269/224429233-c9227e77-1aaf-4320-86f1-6b54227e971a.png">